comm/src/main/scala/coop/rchain/comm/comm.scala

```diff
@@ -0,0 +1,34 @@
+package coop.rchain.comm
+
+import scala.util.Try
+
+trait Comm {
+  def send(data: Seq[Byte], p: PeerNode): Try[Boolean]
+  def recv: Try[Seq[Byte]]
+}
+
+case class NodeIdentifier(pKey: Seq[Byte]) {
```

Comment:
Can you explain why NodeIdentifier is declared as a case class?  It the reason to use it as a value class?

Reply:
Honestly, it's a case class because I was encouraged to use case classes wherever possible in Scala. I guess it's also nice not to have to use `new`.

Reply:
Instead of one-field case classes you can use tagged types or newtype from Shapeless, so it brings typesafety without overhead. Just my two cents.